src/mono/wasm/debugger/BrowserDebugProxy/MemberReferenceResolver.cs

```diff
@@ -242,33 +257,61 @@ async Task<JObject> ResolveAsLocalOrThisMember(string name)
+                JToken objRet = valueOrError.Value.FirstOrDefault(objPropAttr => objPropAttr["name"].Value<string>() == name);
                 if (objRet != null)
                     return await GetValueFromObject(objRet, token);
 
                 return null;
             }
 
-            async Task<JObject> ResolveAsInstanceMember(string expr, JObject baseObject)
+            async Task<JObject> ResolveAsInstanceMember(ArraySegment<string> parts, JObject baseObject, bool hasNullCondition)
             {
+                // if the prevPart has "?" at the end: true
+                bool hasResolvedObjNullCondition = hasNullCondition;
+                bool isPrevPartNull = false;
                 JObject resolvedObject = baseObject;
-                string[] parts = expr.Split('.');
-                for (int i = 0; i < parts.Length; i++)
+                for (int i = 0; i < parts.Count; i++)
                 {
-                    string partTrimmed = parts[i].Trim();
+                    string partTrimmed = parts[i];
                     if (partTrimmed.Length == 0)
                         return null;
 
+                    // current value of resolvedObject is on parts[i - 1]
+                    if (isPrevPartNull || resolvedObject.IsNullValuedObject())
+                    {
+                        // trying null.$member
+                        if (!hasResolvedObjNullCondition)
+                            throw new ReturnAsErrorException($"Cannot access member \"{partTrimmed}\" of a null-valued object.", "ReferenceError");
+
+                        if (i == parts.Count - 1)
+                        {
+                            // this is not ideal, it returns the last object
+                            // that had objectId and was null-valued,
+                            // so the class/description of object are not of the last part
+                            return resolvedObject;
+                        }
+                        // else: trying null?.$member... - allowed operation
+                        isPrevPartNull = true;
+                    }
+
+                    bool hasCurrentPartNullCondition = partTrimmed.Last() == '?';
+                    hasResolvedObjNullCondition = hasCurrentPartNullCondition;
+
+                    // cannot resolve the members but can check if null condition is
+                    // correctly applied and if we should throw or return null-object
                     if (!DotnetObjectId.TryParse(resolvedObject?["objectId"]?.Value<string>(), out DotnetObjectId objectId))
-                        return null;
+                        continue;
 
-                    ValueOrError<GetMembersResult> valueOrError = await proxy.RuntimeGetObjectMembers(sessionId, objectId, null, token);
+                    var args = JObject.FromObject(new { forDebuggerDisplayAttribute = true });
```

Comment:
Is there a test covering this?

Reply:
Yes, it was done to fix existing tests. It's the 1st dot from the PR description: https://github.com/dotnet/runtime/pull/69307#issue-1235045964.